c_emulator/riscv_callbacks.cpp

```diff
@@ -73,7 +74,8 @@ unit freg_write_callback(unsigned reg, uint64_t value)
 {
   // TODO: will only print bits; should we print in floating point format?
   if (config_print_reg) {
-    fprintf(trace_log, "f%d <- 0x%016" PRIX64 "\n", reg, value);
+    // TODO: once the "Q" extension is supported change PRIX64 to PRIX128
```

Comment:
Probably not - if FLEN is >64 bits the callback will be `sail_int` or whatever instead of `uint64_t`. It's going to get awkward for sure...